src/test/pages/admin/userManagement/userPage.ts

```diff
@@ -29,11 +116,201 @@ export default class AdminMenuPage {
         await this.addBtn.click();
         await this.userRole.click();
+        await this.userRoleOption.click();
+        await this.status.click();
+        await this.statusOption.click();
+        await this.employeeName.fill("t");
+        await this.employeeOption.click();
+        await this.usernameField.fill("usernamenttheu");
+        await this.passwordField.fill("admin123");
+        await this.confirmPassword.fill("admin123");
+        await this.submitBtn.click();
+    }
+    async afterCreateEssUser() {
+        await this.successToast.waitFor({ state: 'visible', timeout: 10000 });
+        await expect(this.newEssUser).toBeVisible();
+    }
+    async createAdminUser() {
+        //employee: string, username: string, pass: string, confirm: string
+        await this.adminMenu.click();
+        await this.addBtn.click();
+        await this.userRole.click();
+        await this.adminRoleOption.click();
+        await this.status.click();
+        await this.statusOption.click();
+        await this.employeeName.fill("t");
+        await this.employeeOption.click();
+        await this.usernameField.fill("usernamenttheuAdmin");
+        await this.passwordField.fill("admin123");
+        await this.confirmPassword.fill("admin123");
+        await this.submitBtn.click();
+    }
+    async afterCreateAdminUser() {
+        await this.successToast.waitFor({ state: 'visible', timeout: 10000 });
+        await expect(this.newAdminUser).toBeVisible();
+    }
+    async searchUserName(userName: string) {
+        await this.adminMenu.click();
+        await this.usernameFieldSearch.fill(userName);
+        await this.searchBtn.click();
+    }
+    async afterSearchUserName(checkUser = false) {
+        await this.page.route(`${process.env.SEARCH_URL}`, async (route) => {
+            const response = await route.fetch();
+            expect(response.status()).toBe(200);
+        });
+        //checkUser = true
+        if (checkUser) {
+            await expect(this.newEssUser).toBeVisible();
+            await expect(this.userResult).toHaveCount(1);
+        }
+        //checkUser = false
+        else {
+            await expect(this.notFoundItem).toBeVisible();
+        }
+    }
+    async searchUserRole() {
+        await this.adminMenu.click();
+        await this.userRole.click();
+        await this.adminRoleOption.click();
+        await this.searchBtn.click();
+    }
+    async afterSearchUserRole() {
+        await this.page.waitForTimeout(5000);
+        const results = await this.roleColumnLocator.all();
+        // Check if any results are found
+        expect(results.length).toBeGreaterThan(0);
+        for (let i = 0; i < results.length; i++) {
+            const result = results[i];
+            await expect(result).toBeVisible({ timeout: 5000 });
+            const statusText = await result.textContent();
+            expect(statusText).toBe('Admin');
+        }
+    }
+    async searchEmployeeName() {
```

Comment:
```
async searchEmployeeName(name: string){
    await this.employeeName.fill(name);
}
```

, make sure to enter the full name of the employee you want to search, rather than just typing “t”. The clearer your value is, the better your test case will be.